### PR TITLE
fix: Fix for the test case

### PR DESCRIPTION
**Pull request by sagar-qa007**
## Description
Fix flaky test case.

Fixes # https://app.zenhub.com/workspaces/stability-pod-6690c4814e31602e25cab7fd/issues/gh/appsmithorg/appsmith/37935

## Automation

/ok-to-test tags="@tag.Datasource"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/12174291982>
> Commit: d71af4cb8fa1fb62a9b8977ed872dfa8cdef6fb8
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=12174291982&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Datasource`
> Spec:
> <hr>Thu, 05 Dec 2024 06:59:43 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **Bug Fixes**
	- Updated timeout duration for API calls in the bug test suite to improve test reliability.
	- Shifted focus of test specifications from a template-related test to a bug-related test.

These changes enhance the effectiveness of our testing process, ensuring more accurate results and quicker response times during tests.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/DS_Bug26941_Spec.ts
Comment (yatinappsmith) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/DS_Bug26941_Spec.ts:1]: Raising the wait will resulting in making the test flaky after few days.

Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/DS_Bug26941_Spec.ts:1]: This wait is not a hardcoded sleep; it is the timeout for the API call to respond. Typically, I observe successful responses within 1 second, but for better stability, I have set it to 2 seconds. If the response is received within 1 second, the test will pass without delay.

Maintaining this timeout avoids the need for adding retries, which could mask potential bugs. Introducing retries increases the risk of missing errors and reduces the accuracy of our tests.

**Review by yatinappsmith — APPROVED** [review]

**Review by yatinappsmith — APPROVED** [review]

**Pull request merged**